File: snippets/ts/0027_remove_element.ts
function removeElement(nums: number[], val: number): number {
    let slow: number = 0,
        fast: number = 0;

    while (fast < nums.length) {
        if (nums[fast] !== val) {
            nums[slow++] = nums[fast];
        }
        fast++;
    }
    return slow;
}

const nums: number[] = [3, 2, 2, 3];
const val: number = 3;
console.log(removeElement(nums, val)); // 2

File: src/cpp/0027_remove_element.cc
#include <iostream>
#include <vector>
using namespace std;

// Fast Slow Pointers
int removeElement(vector<int>& nums, int val) {
    size_t n = nums.size();
    size_t slow = 0, fast = 0;

    while (fast < n) {
        if (nums[fast] != val) {
            nums[slow] = nums[fast];
            slow++;
        }
        fast++;
    }
    return (int)slow;
}

int main() {
    vector<int> nums = {3, 2, 2, 3};
    int val = 3;
    cout << removeElement(nums, val) << endl;  // 2
    return 0;
}
